integr8ly/application-monitoring-operator
# Add rest of skeleton for operator

* Add primary resource ApplicationMonitoring api & controller
* Add Makefile & travis.yml
* Add image ref from quay.io

*To verify*

```
oc new-project application-monitoring
oc create -f ./deploy/crds/applicationmonitoring_v1alpha1_applicationmonitoring_crd.yaml
oc create -f ./deploy/crds/applicationmonitoring_v1alpha1_applicationmonitoring_cr.yaml
make code/run
```
Operator should start locally, & logs should have the below 2 entries after a few seconds:

```
{"level":"info","ts":1548349064.0965805,"logger":"controller_applicationmonitoring","caller":"applicationmonitoring/applicationmonitoring_controller.go:84","msg":"Reconciling ApplicationMonitoring","Request.Namespace":"application-monitoring","Request.Name":"example-applicationmonitoring"}
{"level":"info","ts":1548349064.0966141,"logger":"controller_applicationmonitoring","caller":"applicationmonitoring/applicationmonitoring_controller.go:100","msg":"TODO: Reconcile implementation","Request.Namespace":"application-monitoring","Request.Name":"example-applicationmonitoring"}
```

*TODO*

* in separate follow-up PR - reconcile implementation for installing the application monitoring stack
